queries/faust/highlights.scm

```diff
@@ -0,0 +1,216 @@
+; Identifiers
+(identifier) @variable
+
+(process) @variable.builtin
+
+(parameters
+  (identifier)) @variable.parameter
+
+; Literals
+(_
+  filename: (string)) @string.special.path
+
+(function_names) @string.special
+
+(documentation) @string.documentation @spell
+
+[
+  (string)
+  (fstring)
+] @string
+
+(real) @number.float
+
+[
+  (int)
+  (real)
+] @number
+
+; Types
+(global_metadata
+  key: (identifier)) @property
+
+(function_metadata
+  key: (identifier)) @property
+
+(_
+  type:
+    [
+      (int)
+      (float)
+      (any)
+    ]) @type.builtin
+
+[
+  (single_precision)
+  (double_precision)
+  (quad_precision)
+  (fixed_point_precision)
+] @attribute
+
+; Functions
+(function_definition
+  name: (identifier) @function)
+
+[
+  (lambda)
+  (prefix)
+  (prim1)
+  (prim2)
+  (prim3)
+  (prim4)
+  (prim5)
+  (function_call)
+] @function.call
+
+[
+  (exp)
+  (log)
+  (log10)
+  (sqrt)
+  (abs)
+  (floor)
+  (ceil)
+  (rint)
+  (round)
+  (acos)
+  (asin)
+  (atan)
+  (cos)
+  (sin)
+  (tan)
+  (atan2)
+  (int_cast)
+  (float_cast)
+  (pow_fun)
+  (min)
+  (max)
+  (fmod)
+  (remainder)
+  (prefix_prim)
+  (attach)
+  (enable)
+  (control)
+  (rdtable)
+  (rwtable)
+  (select2)
+  (select3)
+  (component)
+  (library)
+  (lowest)
+  (highest)
+  (assertbounds)
+  (par)
+  (seq)
+  (sum)
+  (prod)
+  (vslider_type)
+  (hslider_type)
+  (nentry_type)
+  (vbargraph_type)
+  (hbargraph_type)
+  (vgroup_type)
+  (hgroup_type)
+  (tgroup_type)
+  "button"
+  "checkbox"
+  "soundfile"
+  "inputs"
+  "outputs"
+  "route"
+] @function.builtin
+
+; xor is a @keyword.operator
+[
+  (add)
+  (sub)
+  (mult)
+  (div)
+  (mod)
+  (pow)
+  (or)
+  (and)
+  (lshift)
+  (rshift)
+  (lt)
+  (le)
+  (gt)
+  (ge)
+  (eq)
+  (neq)
+  (delay)
+  (one_sample_delay)
+  "="
+  "=>"
+  "->"
+] @operator
+
+(recursive
+  "~" @operator)
+
+(sequential
+  ":" @operator)
+
+(split
+  "<:" @operator)
+
+(merge
+  ":>" @operator)
+
+(parallel
+  "," @operator)
+
+; Keywords
+[
+  (par)
+  (seq)
+  (sum)
+  (prod)
+] @keyword.repeat
+
+(file_import
+  "import") @keyword.import
+
+[
+  (wire)
+  (cut)
+  (mem)
+  "declare"
+  "with"
+  "environment"
+  "case"
+  "ffunction"
+  "fconstant"
+  "fvariable"
+] @keyword
+
+(xor) @keyword.operator
+
+; Punctuation
+[
+  ","
+  ";"
+  "."
+] @punctuation.delimiter
+
+[
+  "("
+  ")"
+  "["
+  "]"
+  "{"
+  "}"
+] @punctuation.bracket
+
+; Comments
+(comment) @comment @spell
+
+[
+  (doc_metadata)
+  (doc_equation)
+  (doc_diagram)
+  (doc_listing)
+  (doc_notice)
+] @tag
```

Comment:
You should only capture the tags and not their contents.

Reply:
Thanks - I'll just skip tags for now

Reply:
I mean, if you only have a subset of tags, it's better to just highlight those instead of injecting xml.

Reply:
If it's frowned upon to inject xml parsing for subsets of the language that are defined in XML if only a handful of tags are expected, I'll just remove the XML injection for now and the tag highlight names for now.

Reply:
It's not frowned upon per se, but users would have to install a parser they may not otherwise need.
You can remove the injection but keep the highlights (just apply them to the actual tags).

Reply:
I see, so it's a requirement, not optional, to have the injected language installed. Ok then definitely best not to have it. I'll resolve this then.

Reply:
> I see, so it's a requirement, not optional, to have the injected language installed.

Well, if it's not installed, it won't be highlighted.